core/src/impl/Kokkos_Atomic_Generic.hpp

```diff
@@ -56,12 +56,44 @@
 namespace Kokkos {
 namespace Impl {
 
+template <class Op, class Scalar1, class Scalar2, class Enable = bool>
+struct _check_early_exit_impl {
+  KOKKOS_FORCEINLINE_FUNCTION
+  static constexpr bool check(Op const&, Scalar1 const&,
+                              Scalar2 const&) noexcept {
+    return false;
+  }
+};
+
+template <class Op, class Scalar1, class Scalar2>
+struct _check_early_exit_impl<
+    Op, Scalar1, Scalar2,
+    decltype(bool(std::declval<Op const&>().check_early_exit(
+        std::declval<Scalar1 const&>(), std::declval<Scalar2 const&>())))> {
```

Comment:
```suggestion
    decltype(std::declval<Op const&>().check_early_exit(
        std::declval<Scalar1 const&>(), std::declval<Scalar2 const&>()))> {
```
PS format may be wrong